src/runtime/dict.cpp

```diff
@@ -397,6 +400,18 @@ Box* dictDelitem(BoxedDict* self, Box* k) {
     return None;
 }
 
+// Analoguous to CPython's, used for sq_ slots.
+static int dict_ass_sub(PyDictObject* mp, PyObject* v, PyObject* w) {
+    Box* res;
+    if (w == NULL) {
+        res = dictDelitem((BoxedDict*)mp, v);
+    } else {
+        res = dictSetitem((BoxedDict*)mp, v, w);
```

Comment:
We should wrap these in a try-catch-setCapiException.  it's also nice to put a "noexcept" specifier on the method as a whole, since that will get the compiler to enforce that the function doesn't end up throwing anything (by adding try-catch-std::terminate() wrappers).